### PR TITLE
Close the master pty before terminating with SIGTERM.

### PR DESCRIPTION
**Pull request by jsm222**
See discussion here https://github.com/lxqt/qtermwidget/pull/171#issuecomment-392157569.
Only tested with csh on FreeBSD.

### CODE REVIEW

File: lib/kptyprocess.cpp [hunk 1]
```diff
@@ -72,13 +72,14 @@ KPtyProcess::~KPtyProcess()
...
         }
 
-        qWarning() << Q_FUNC_INFO << "the terminal process is still running, trying to stop it by SIGHUP";
-        ::kill(pid(), SIGHUP);
-        waitForFinished(300);
-        if (state() != QProcess::NotRunning)
-            qCritical() << Q_FUNC_INFO << "process didn't stop upon SIGHUP and will be SIGKILL-ed";
+        d->pty->close();
+        terminate();
+        if(waitForFinished(300)) {
```
Comment (ypnos) [inline lib/kptyprocess.cpp]: This looks odd. Do the same thing in both cases?

**Comment by jsm222** [comment]
Yes I did, but how does it close the master pty?

**Comment by palinek** [comment]
> Yes I did, but how does it close the master pty?

Did you look into the d-tor of `KPtyDevice`?
```cpp
KPtyDevice::~KPtyDevice()
{
    close();
}
```

**Comment by jsm222** [comment]
Ok. will close this one

**Pull request closed without merging**